src/Transport/Topology/Topologies/ForwardingTransportInfrastructure.cs

```diff
@@ -0,0 +1,47 @@
+namespace NServiceBus
+{
+    using System.Threading;
+    using Settings;
+    using Transport.AzureServiceBus;
+
+    class ForwardingTransportInfrastructure : AzureServiceBusTransportInfrastructure
+    {
+        ForwardingTopologySectionManager topologySectionManager;
+        string bundlePrefix;
+        volatile int initializeSignaled;
+
+        public ForwardingTransportInfrastructure(SettingsHolder settings) : base(settings)
+        {
+            settings.SetDefault(WellKnownConfigurationKeys.Topology.Bundling.NumberOfEntitiesInBundle, 1);
+            settings.SetDefault(WellKnownConfigurationKeys.Topology.Bundling.BundlePrefix, "bundle-");
+        }
+
+        protected override ITopologySectionManagerInternal CreateTopologySectionManager(string defaultAlias, NamespaceConfigurations namespaces, INamespacePartitioningStrategy partitioning, AddressingLogic addressing)
+        {
+            var endpointName = Settings.EndpointName();
+            var numberOfEntitiesInBundle = Settings.Get<int>(WellKnownConfigurationKeys.Topology.Bundling.NumberOfEntitiesInBundle);
+            bundlePrefix = Settings.Get<string>(WellKnownConfigurationKeys.Topology.Bundling.BundlePrefix);
+
+            topologySectionManager = new ForwardingTopologySectionManager(defaultAlias, namespaces, endpointName, numberOfEntitiesInBundle, bundlePrefix, partitioning, addressing);
+            // utter hack because we require this at resource creation time
+            topologySectionManager.Initialize = async () =>
+            {
+                if (Interlocked.Exchange(ref initializeSignaled, 1) != 0)
+                {
+                    return;
+                }
+
+                var bundleConfigurations = await NumberOfTopicsInBundleCheck.Run(namespaceManager, namespaceConfigurations, bundlePrefix)
+                    .ConfigureAwait(false);
+
+                topologySectionManager.BundleConfigurations = bundleConfigurations;
+            };
```

Comment:
Why not just pass namespaceManager and namespaceConfigurations in the constructor and have this lambda initialized inside the section manager?

Reply:
The section manager is not async. Previously we wrapped this call in a Lazy and executed it with GetAwaiter().GetResult(). In the past we had a discussion that the section manager could become async but if I recall correctly you rightfully said that building the topology should be possible local only without information from ASB itself. Therefore this bundle check does not belong into the manager.

Reply:
Ah yes... add this comment in the code instead of //utter hack... so I wont forget next time I see it ;)